File: Selenium/Activities/Activity8.py
from selenium import webdriver
from selenium.webdriver.common.by import By
with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/mouse-events")
    print(f"the title of the main page is: {driver.title}")

    actions = webdriver.ActionChains(driver)
    cargoLock = driver.find_element(By.XPATH,"//h1[text()='Cargo.lock']")
    cargotoml = driver.find_element(By.XPATH,"//h1[text()='Cargo.toml']")
    src = driver.find_element(By.XPATH,"//h1[text()='src']")
    target = driver.find_element(By.XPATH,"//h1[text()='target']")

    actions.click(cargoLock).pause(5).move_to_element(cargotoml).pause(5).click(cargotoml).perform()

    result = driver.find_element(By.ID,"result")
    print(f"the message is: {result.text}")

    actions.double_click(src).pause(5).context_click(target).pause(5).perform()
    actions.click(driver.find_element(By.XPATH,"//div[@id='menu']/div/ul/li[1]")).pause(5).perform()

    result = driver.find_element(By.ID,"result")
    print(f"the message is: {result.text}")
    driver.quit()
		

File: Selenium/Activities/Activity8.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Action;
import org.openqa.selenium.interactions.Actions;

public class Activity8 {
	public static void main(String args[]) {
		WebDriver driver = new FirefoxDriver();
		driver.get("https://training-support.net/webelements/mouse-events");
		System.out.println("The title of the main page is: "+driver.getTitle());
		
		Actions builder = new Actions(driver);
		

		WebElement cargoLock = driver.findElement(By.xpath("//h1[text()='Cargo.lock']"));
		WebElement cargotoml = driver.findElement(By.xpath("//h1[text()='Cargo.toml']"));
		WebElement src = driver.findElement(By.xpath("//h1[text()='src']"));
		WebElement target = driver.findElement(By.xpath("//h1[text()='target']"));
		
		Action actionsToPerform1 =builder.click(cargoLock).pause(1000).moveToElement(cargotoml).pause(1000).click(cargotoml).build();
		
		actionsToPerform1.perform();
		
		WebElement result = driver.findElement(By.id("result"));
		System.out.println("the message is: "+result.getText());
		
		Action actionsToPerform2 = builder.doubleClick(src).pause(1000).contextClick(target).pause(1000).build();
		
		actionsToPerform2.perform();
		
		builder.click(driver.findElement(By.xpath("//div[@id='menu']/div/ul/li[1]"))).pause(1000).build().perform();
		System.out.println("the message is: "+result.getText());
		
		driver.quit();
	}
		
}
